scripts/markdown-code-checker.py

```diff
@@ -181,6 +203,14 @@ def parse_args():
         print("[error] This path doesn't exist")
         sys.exit(-1)
 
+    if (
+        args.syntax_check
+        and "cf3" in args.languages
+        and not which("/var/cfengine/bin/cf-promises")
```

Comment:
Not a big concern in this tool, but be aware that when you do this (first check if a file / program exists, and then later open it, assuming it still exists) you open up for the possibility of that program being modified or deleted in between. This is called a [Time-of-check to time-of-use race condition](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use).

Generally, the preferred pattern, especially in Python, is to just try to open it and then handle the exceptions if there's something wrong. Or you can do both, check early as you do here, to give a nice and early error message, but then still handle the exceptions correctly when you open `cf-promises`, because it could have disappeared in the meantime.